pkg/virt-handler/vm.go

```diff
@@ -1904,15 +1904,53 @@ func (d *VirtualMachineController) isPreMigrationTarget(vmi *v1.VirtualMachineIn
 }
 
 func (d *VirtualMachineController) checkNetworkInterfacesForMigration(vmi *v1.VirtualMachineInstance) error {
-	networks := map[string]*v1.Network{}
-	for _, network := range vmi.Spec.Networks {
-		networks[network.Name] = network.DeepCopy()
+	err := validatePodNetworkInterfaceUsesMasqueradeBinding(vmi)
+	if err != nil {
+		return err
 	}
+
+	err = d.validateSRIOVInterfacesForMigration(vmi)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validatePodNetworkInterfaceUsesMasqueradeBinding(vmi *v1.VirtualMachineInstance) error {
+	interfacesByName := map[string]v1.Interface{}
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Masquerade == nil && networks[iface.Name].Pod != nil {
-			return fmt.Errorf("cannot migrate VMI which does not use masquerade to connect to the pod network")
+		interfacesByName[iface.Name] = iface
+	}
+
+	vmiPodNetworkName := lookupVMIPodNetworkName(vmi.Spec.Networks)
+	if vmiPodNetworkName != "" && interfacesByName[vmiPodNetworkName].Masquerade == nil {
+		return fmt.Errorf("cannot migrate VMI which does not use masquerade to connect to the pod network")
+	}
+
+	return nil
+}
+
+func lookupVMIPodNetworkName(networks []v1.Network) string {
+	for _, network := range networks {
+		if network.Pod != nil {
+			return network.Name
+		}
+	}
+
+	return ""
+}
+
+func (d *VirtualMachineController) validateSRIOVInterfacesForMigration(vmi *v1.VirtualMachineInstance) error {
```

Comment:
I think you could simply use the `IsSRIOVVmi(vmi)` from pkg/util and it'll simply be 

if util.IsSRIOVVmi(vmi) && !d.clusterConfig.SRIOVLiveMigrationEnabled() {
				return fmt.Errorf("Live Migration of guests with SR-IOV interfaces is not supported")
			}

Reply:
My intent was to make it easier to understand by moving the logic (that checks that the VMI has SRIOV iface and the feaute-gate is enabled) to a function that describe what it does instead of adding a comment.